### ISSUE TITLE
Requests with unsupported version don't return 4xx error code

### ISSUE THREAD

**Issue by heorhi-kaubaska** [dotnet/aspnet-api-versioning#918]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Request with any `api-version` is handled by controller and there are no `api-supported-versions` header in response despite `ReportApiVersions` is set to `true`.
I used the [Basic example](https://github.com/dotnet/aspnet-api-versioning/tree/main/examples/AspNetCore/WebApi/BasicExample) with slight changes to reproduce the issue.
A successful status code is returned no matter which version is sent:
✔️ v1.0
![image](https://user-images.githubusercontent.com/68230484/202671330-4de481e0-ce67-4abc-97fd-b2f55c96dfac.png)
❌ v2.0
![image](https://user-images.githubusercontent.com/68230484/202671473-a28c71b5-f849-470a-8c7b-3e9b353cad50.png)
❌ v9999
![image](https://user-images.githubusercontent.com/68230484/202671513-a9353a86-0cea-419e-85ab-8475f81d577e.png)

### Expected Behavior

Request with unsupported version should return 4xx status code

### Steps To Reproduce

.csproj file
```
<Project Sdk="Microsoft.NET.Sdk.Web">
  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <Nullable>enable</Nullable>
    <ImplicitUsings>enable</ImplicitUsings>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Asp.Versioning.Mvc" Version="6.2.1" />
  </ItemGroup>
</Project>
```
Program.cs
```
var builder = WebApplication.CreateBuilder( args );

// Add services to the container.

builder.Services.AddControllers();
builder.Services.AddApiVersioning(
        options =>
        {
            // reporting api versions will return the headers
            // "api-supported-versions" and "api-deprecated-versions"
            options.ReportApiVersions = true;
        } )
    .AddMvc();

var app = builder.Build();

// Configure the HTTP request pipeline.

app.UseHttpsRedirection();
app.UseAuthorization();
app.MapControllers();
app.Run();
```
TestController.cs
```
using Asp.Versioning;
using Microsoft.AspNetCore.Mvc;

namespace ApiVersioningTest;

[ApiVersion("1.0")]
[Route( "api/v{version:apiVersion}/test" )]
public class TestController : ControllerBase
{
    [HttpGet("version")]
    public IActionResult Versioned(ApiVersion apiVersion)
    {
        return Ok(apiVersion);
    }
}
```

### .NET Version

6.0.301

**Comment by commonsensesoftware** [comment]
The first problem is because `TestController` is not actually an _API controller. It's missing `[ApiController]`. The attribute can also be applied on the entire assembly, but I don't see that and most people don't realize you can do that. Since `TestController` is not treated as an API controller, `[ApiVersion]` has no meaning; it's pure metadata. The `apiVersion` route constraint will allow any valid `ApiVersion`.

Reporting supported and deprecated API versions is dependent on getting to a candidate to use its metadata. Necessary changes in the routing policy _may_ now eliminate candidates before there's any chance to retrieve metadata. This currently makes it impossible to report anything in some cases - mostly notably when something _could_ exist, but doesn't. This change was needed to fix other issues related to `405`, `406`, and `415`. This change was called out months ago in the roadmap and release notes. #876 is tracking a similar request. This is a difficult problem to solve. The only thing to go by in the request is the API version, but different APIs can have overlapping versions that don't necessarily have the version sets.

Since you're versioning by URL segment (the un-RESTful way), there was also a fix in `Asp.Versioning.Http` `6.2.2` that addresses a scenario where you'll get `400` instead of `404`. The URL _is_ the identifier so if no API versions are detected in it, then it _should_ be treated as `404`.

**Comment by commonsensesoftware** [comment]
This is fixed in `6.3.x` and `7.0.0-*`
